src/pages/ParticipantsPage.js

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import {
+    View,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import styles from '../styles/styles';
+import ONYXKEYS from '../ONYXKEYS';
+import HeaderWithCloseButton from '../components/HeaderWithCloseButton';
+import Navigation from '../libs/Navigation/Navigation';
+import ScreenWrapper from '../components/ScreenWrapper';
+import OptionsList from '../components/OptionsList';
+import ROUTES from '../ROUTES';
+import CONST from '../CONST';
+
+const matchType = PropTypes.shape({
+    params: PropTypes.shape({
+        // reportID passed via route /participants/:reportID
+        reportID: PropTypes.string,
+    }),
+});
+
+const myPersonalDetailsType = PropTypes.shape({
+    // Display name of the current user from their personal details
+    displayName: PropTypes.string,
+
+    // Avatar URL of the current user from their personal details
+    avatar: PropTypes.string,
+
+    // login of the current user from their personal details
+    login: PropTypes.string,
+});
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: myPersonalDetailsType.isRequired,
+
+    // List of reports
+    reports: PropTypes.shape({
+        icons: PropTypes.arrayOf(PropTypes.string),
+        reportName: PropTypes.string,
+        participants: PropTypes.arrayOf(PropTypes.string),
+    }).isRequired,
+
+    // Route params
+    route: matchType.isRequired,
+
+    priorityMode: PropTypes.string.isRequired,
+};
+
+const getAllParticipants = (report, personalDetail) => {
+    const {icons, participants, reportName} = report;
+    const displayNames = reportName.split(', ');
+
+    const otherMembers = participants.map((login, index) => (
+        {
+            alternateText: login,
+            icons: [icons[index]],
+            keyForList: `${index}`,
+            tooltipText: login,
+            text: displayNames[index],
+            login,
+            displayName: displayNames[index],
+        }
+    ));
+
+    const final = [...otherMembers,
+        {
+            icons: [personalDetail.avatar],
+            text: personalDetail.displayName,
+            login: personalDetail.login,
+            alternateText: personalDetail.login,
+            keyForList: `${otherMembers.length}`,
+            tooltipText: personalDetail.login,
+            displayName: personalDetail.displayName,
+        },
+    ];
+
+    return final.map(item => ({...item, participantsList: [item]}));
```

Comment:
Why are we adding the `participantsList` key?

Reply:
The `OptionsList` component seems to use the text to display from within the `participantsList` key. It was pretty weird for me too, but it threw error without it.

Reply:
Hey @npsedhain thanks for your patience while I was looking into this. Here were my concerns:
1. We're sending two copies of the same data, nested in itself
1. We're putting a single object into an array just to satisfy a proptype rule - which to me indicates that we're holding something wrong. 

These two concerns acted as a red flag for me, indicating that we were holding something wrong. 

I started my investigation by looking at the other place in which `OptionsList` is used. The items that are usually in `sections.data` are not single people but whole reports (chats). While Options list _looks_ like the correct component for the job, it was actually written with a different purpose in mind. 

Two potential avenues forward:
1. Don't worry about it. It works so whatever.
2. Write a component that's actually meant for this. 

To help us suss this out, I'd like to include @marcaaron in this decision. He wrote much of `OptionsList` and is very knowledgeable in the codebase.

Reply:
Thanks, @Luke9389, I will the changes as suggested by @marcaaron, i.e., only sending the two keys needed.